addons/mail/static/src/core/common/composer_model.js

```diff
@@ -13,6 +13,16 @@ export class Composer extends Record {
         return super.insert(...arguments);
     }
 
+    clear() {
+        this.attachments.length = 0;
+        this.textInputContent = "";
+        Object.assign(this.selection, {
+            start: 0,
+            end: 0,
+            direction: "none",
+        });
```

Comment:
```suggestion
        this.selection = {
            start: 0,
            end: 0,
            direction: "none",
        };
```

Reply:
Or better, const the default value outside the class and use it for both cleaning and initialisation :)

Reply:
IMO `Object.assign()` approach is simplest and safest, as any added field to `selection` does not require to adapt these LOCs.